src/js/components/modals/UpgradePackageModal.js

```diff
@@ -0,0 +1,126 @@
+import {Dropdown, Modal} from 'reactjs-components';
+import React from 'react';
+
+const METHODS_TO_BIND = ['handleUpgradeStart', 'handleVersionSelection'];
+
+class UpgradePackageModal extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      upgradeVersion: null
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleUpgradeStart() {
+    let upgradeVersion = this.state.upgradeVersion;
+    if (!upgradeVersion) {
+      upgradeVersion = this.getLatestVersion(this.props.cosmosPackage);
+    }
+
+    console.log(`starting upgrade to ${upgradeVersion}`);
+  }
+
+  handleVersionSelection(version) {
+    this.setState({upgradeVersion: version.id});
+  }
+
+  getLatestVersion(cosmosPackage) {
+    let availableVersions = cosmosPackage.getUpgradeVersions();
+    return availableVersions[availableVersions.length - 1];
+  }
+
+  getVersionDropdownItems(availableVersions) {
+    return availableVersions.map(function (version) {
+      return {id: version, html: `Version ${version}`};
+    });
+  }
+
+  getModalContents() {
+    if (!this.props.open) {
+      return null;
+    }
+
+    let {cosmosPackage, packageName, packageVersion} = this.props;
+
+    return (
+      <div>
+        <div className="modal-content">
+          <div className="modal-content-inner container container-pod
+            container-pod-short horizontal-center">
+            <div className="icon icon-jumbo icon-image-container
+              icon-app-container">
+              <img src={cosmosPackage.getIcons()['icon-large']} />
+            </div>
+            <h2 className="short">{packageName}</h2>
+            <p className="flush">
+              {packageName} {packageVersion}
+            </p>
+          </div>
+        </div>
+        <div className="modal-footer">
```

Comment:
is there a reason why we're going outside of using RJSC's Modal's footer prop? (we do this in other places, but because we have to)